screens/dashboard/components/upcoming.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import {
+  StyleSheet,
+  Text,
+  View,
+  TouchableHighlight,
+} from 'react-native';
+import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import moment from 'moment';
+
+// import global actions
+import * as actions from '../actions';
+
+// import global styles
+// @TODO: move the global styles from this screen into ../../styles/common 
+import commonStyles from '../../../styles/common';
+const styles = StyleSheet.create(commonStyles);
+
+type Props = {
+  actions: Object,
+  profile: Object,
+  navigation: Object
+};
+
+
+export class Upcoming extends Component<Props> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false
+    };
+    this.icons = {
+      'open': 'angle-down',
+      'close': 'angle-up'
+    };
+  }
+
+
+  showUpcoming(firstOrRemaining) {
+
+    let upcomingArray = ((this.props.profile || {}).upcomingArray || {});
```

Comment:
you're null checking (good) but the default your return is an empty object.  It should be an empty array.

Reply:
`      const upcomingArray = ((this.props.profile || []).upcomingArray || []);
`